File: examples/example-kubernetes-client-openapi-informer-python/src/main/python/micronaut/client/configmap/ConfigMapController.py
from micronaut.context.annotation import Requires
from micronaut.context.env import Environment
from micronaut.http.annotation import Controller, Get
from micronaut.kubernetes.client.openapi.informer import SharedIndexInformerFactory
from micronaut.kubernetes.client.openapi.model import V1ConfigMap
from micronaut.kubernetes.client.openapi.resolver import NamespaceResolver


@Requires(env=Environment.KUBERNETES)
@Controller("config-maps")
class ConfigMapController:

    def __init__(self, shared_index_informer_factory: SharedIndexInformerFactory, namespace_resolver: NamespaceResolver):
        self.shared_index_informer_factory = shared_index_informer_factory
        self.namespace_resolver = namespace_resolver

    @Get
    # tag::getAll[]
    def all(self) -> list[V1ConfigMap]:
        namespace = self.namespace_resolver.resolveNamespace()
        informer = self.shared_index_informer_factory.getExistingSharedIndexInformer(
            V1ConfigMap,
            namespace)
        indexer = informer.getIndexer()
        return indexer.list()
    # end::getAll[]

    @Get("/{name}")
    def config_map(self, name: str) -> V1ConfigMap | None:
        namespace = self.namespace_resolver.resolveNamespace()
        # tag::get[]
        informer = self.shared_index_informer_factory.getExistingSharedIndexInformer(
            V1ConfigMap,
            namespace)
        # end::get[]
        indexer = informer.getIndexer()
        return indexer.getByKey(namespace + "/" + name)

File: examples/example-kubernetes-client-openapi-informer-kotlin/src/main/kotlin/micronaut/client/configmap/ConfigMapController.kt
package micronaut.client.configmap

import io.micronaut.context.annotation.Requires
import io.micronaut.context.env.Environment
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.kubernetes.client.openapi.informer.SharedIndexInformerFactory
import io.micronaut.kubernetes.client.openapi.model.V1ConfigMap
import io.micronaut.kubernetes.client.openapi.resolver.NamespaceResolver

@Requires(env = [Environment.KUBERNETES])
@Controller("config-maps")
class ConfigMapController(
    private val sharedIndexInformerFactory: SharedIndexInformerFactory,
    private val namespaceResolver: NamespaceResolver
) {

    @Get
    //tag::getAll[]
    fun all(): List<V1ConfigMap> {
        val namespace = namespaceResolver.resolveNamespace()
        val informer = sharedIndexInformerFactory.getExistingSharedIndexInformer(
            V1ConfigMap::class.java,
            namespace)
        val indexer = informer!!.indexer
        return indexer.list()
    }
    //end::getAll[]

    @Get("/{name}")
    fun configMap(name: String): V1ConfigMap? {
        val namespace = namespaceResolver.resolveNamespace()
        //tag::get[]
        val informer = sharedIndexInformerFactory.getExistingSharedIndexInformer(
            V1ConfigMap::class.java,
            namespace)
        //end::get[]
        val indexer = informer!!.indexer
        return indexer.getByKey("$namespace/$name")
    }
}
